### PR TITLE
[New Extension] Send To Memos

### PR DESCRIPTION
**Pull request by JakeLaoyu**
## Description

Raycast extension for [usememos/memos](https://github.com/usememos/memos).

## Screencast

SCR-20230210-x5j
SCR-20230210-x6a

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by JakeLaoyu** [comment]
Why prettier lint can't pass?

File: extensions/send-to-memos/src/sendMemoForm.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    popToRoot({ clearSearchBar: true });
+  }
+
+  return createdMarkdown ? (
+    <Detail
+      markdown={createdMarkdown}
+      actions={
+        createdUrl && (
+          <ActionPanel>
+            <Action title="Open In Browser" onAction={openWeb} />
```
Comment (pernielsentikaer) [inline extensions/send-to-memos/src/sendMemoForm.tsx:116]:
```suggestion
            <Action title="Open in Browser" onAction={openWeb} />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension and I have some feedback ready for you:

* Add a placeholder/description of where to get the Open API
image

* `Open Memo Website` opens a folder, is that as expected? I didn't set it up 😊

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by JakeLaoyu** [comment]
@pernielsentikaer Thank you for feedback. I have changed.
You can get Open Api from [memos demo](https://demo.usememos.com/).

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

You could use [useFetch](https://developers.raycast.com/utilities/react-hooks/usefetch) from `@raycast/api` to cache t he initial data, this way your extension will be much faster 😊

**Comment by JakeLaoyu** [comment]
> Hi 👋
> 
> You could use [useFetch](https://developers.raycast.com/utilities/react-hooks/usefetch) from `@raycast/api` to cache t he initial data, this way your extension will be much faster 😊

Okay, I will add caches for GET requests such as lists and tags.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
* Currently, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

image

* When submitting `Send MEMO` I get this error

Raycast-v4UZ6gNf@2x

```
20:46:44.723 
DataCloneError: function transformRequest(data, headers) {
    const contentType = headers.getContentType() || "";
    const ha...<omitted>... } could not be cloned.

    const contentType = headers.getContentType() || "";
DOMException:domexception:53:5
MessagePort.<anonymous>:index.js:1:1984
``

**Comment by JakeLaoyu** [comment]
- I have changed to limit to  US English.

- This is so bad about the error. The official test demo disabled the post request.  I deployed a demo project on my server. Maybe you can test it with the account below.

openApi: https://memo-test.jakeyu.top/api/memo?openId=408c75c0-ef46-4072-802e-c65b247105dc
website: https://memo-test.jakeyu.top
username: test
password: test

**Comment by pernielsentikaer** [comment]
Hi 👋 

Thanks for setting it up so I can test it; however, it seems like something is wrong with the implementation

Raycast-wtDvsUGP@2x

**Comment by JakeLaoyu** [comment]
> Hi 👋
> 
> Thanks for setting it up so I can test it; however, it seems like something is wrong with the implementation
> 
> Raycast-wtDvsUGP@2x

Maybe it's because my server is in China. I'll find another way.

**Comment by pernielsentikaer** [comment]
Let's just continue without, let's make a new PR if anything is reported instead 😊

**Comment by JakeLaoyu** [comment]
> Let's just continue without, let's make a new PR if anything is reported instead 😊

That's pretty good. THX.

**Comment by JakeLaoyu** [comment]
Hi @pernielsentikaer , are there still issues that I need to fix?

**Comment by JakeLaoyu** [comment]
> 

Of course. I thought so, too.
Originally, I just wanted to make a send command, but then I wrote more and more.

**Review by pernielsentikaer — APPROVED** [review]
Hello 👋

I gave it a quick spin, works as expected, approved 🔥

**Comment by JakeLaoyu** [comment]
> Hello 👋
> 
> I gave it a quick spin, works as expected, approved 🔥

THX

**Pull request merged**